ecoscope/mapping/map.py

```diff
@@ -478,302 +404,59 @@ def add_pil_image(self, image, bounds, name=None, zoom=True, opacity=1):
-    imagePath: str
-        Path to the image file for the north arrow. If not provided, default SVG image is uploaded.
-    position : str
-        Possible values are 'topleft', 'topright', 'bottomleft' or 'bottomright'.
-    angle : int
-        Angle for the north arrow to be rotated.
-    scale : float
-        Scale dimensions of north arrow.
-
-    """
-
-    _template = Template(
-        """
-        {% macro script(this, kwargs) %}
-        var {{ this.get_name() }} = L.Control.extend({
-            onAdd: function(map) {
-                var control;
-                if (this.options.imagePath){
-                    var img = document.createElement('img');
-                    img.src=this.options.imagePath;
-                    img.style.width = (this.options.scale * 100) + 'px';
-                    img.style.height = (this.options.scale * 100) + 'px';
-                    control=img;
-                } else {
-                    var template = document.createElement('template');
-                    template.innerHTML = `{{ this.html }}`.trim();
-                    control = template.content.firstChild
-                }
-
-                control.style.transform = 'rotate({{ this.angle }}deg)';
-                return control;
-            }
-        });
-        (new {{ this.get_name() }}({{ this.options|tojson }})).addTo({{this._parent.get_name()}});
-
-        {% endmacro %}
-    """
-    )
-
-    def __init__(self, html, imagePath="", position="bottomright", angle="", scale=""):
-        super().__init__(html=html, position=position)
-        self.angle = angle
-        self.options = folium.utilities.parse_options(imagePath=imagePath, scale=scale, position=position)
-
-
-class ScaleElement(MacroElement):
-    """
-    Class to wrap arbitrary HTML as Leaflet Control.
-
-    Parameters
-    ----------
-    html : str
-        HTML to render an element from.
-    position : str
-        Possible values are 'topleft', 'topright', 'bottomleft' or 'bottomright'.
-
-    """
-
-    _template = Template(
-        """
-        {% macro script(this, kwargs) %}
-
-        var {{ this.get_name() }} = L.Control.Scale.extend({
-            onAdd: function(map) {
-
-                var container = document.createElement('div');
-                container.classList.add("leaflet-control-scale");
-                container.innerHTML = `{{ this.html }}`.trim();
-
-                this._scale = container.firstChild;
-
-                map.on('move', this._update, this);
-                map.whenReady(this._update, this);
-
-                return container;
+        data = BytesIO()
+        image.save(data, "PNG")
+
+        url = "data:image/png;base64," + base64.b64encode(data.getvalue()).decode("utf-8")
+        layer = BitmapLayer(image=url, bounds=bounds.tolist(), opacity=opacity)
+        self.add_layer(layer, zoom=zoom)
+
+    @staticmethod
+    def get_named_tile_layer(layer: str) -> BitmapTileLayer:
```

Comment:
I think eventually we should adopt this library to provide us with named XYZ tile services that we can support: https://github.com/geopandas/xyzservices

Reply:
I agree - the main lift here would be adding wms layer support to lonboard